lib/balance-tx/test/spec/Internal/Cardano/Write/Tx/BalanceSpec.hs

```diff
@@ -2688,19 +2690,13 @@ instance forall era. IsRecentEra era => Arbitrary (PartialTx era) where
             (fromCardanoApiUTxO inputUTxO)
             (redeemers)
             mempty
-    shrink (PartialTx tx inputUTxO redeemers timelockKeyWitnessCounts) =
-        [ PartialTx tx inputUTxO' redeemers timelockKeyWitnessCounts
-        | inputUTxO' <- shrinkInputResolution inputUTxO
+    shrink partialTx@PartialTx {tx, extraUTxO} =
+        [ partialTx {extraUTxO = extraUTxO'}
+        | extraUTxO' <- shrinkInputResolution extraUTxO
```

Comment:
Here we avoid pattern matching on fields of `PartialTx` that are unaffected by shrinking.